tests/foreman/api/test_contentview.py

```diff
@@ -1173,25 +1173,25 @@ def test_negative_readonly_user_actions(
     with pytest.raises(HTTPError):
         target_sat.api.ContentView(server_config=cfg, organization=module_org).create()
     # Check that we can read our content view with custom user
-    content_view = target_sat.api.ContentView(server_config=cfg, id=content_view.id).read()
+    content_view_readonly_user = target_sat.api.ContentView(server_config=cfg, id=content_view.id)
```

Comment:
Based on the comment here, we should still attempt to read the content view as that is the only thing this user should be able to do.

Reply:
I think this needs to be fixed in the nailgun as the nailgun does not return the correct auth details, It is been reset to default auth details even though when the custom auth details are passed.

Reply:
While me and @vijaysawant debug this further https://github.com/SatelliteQE/nailgun/blob/master/nailgun/entity_mixins.py#L822 line where we are getting type error for doing read call over target_sat (using Dec Class) which results in setting default nailgun object and that is returned